plugins/module_utils/foundation/image_nodes.py

```diff
@@ -265,6 +298,26 @@ def _get_cvm_ip(self, node):
             raise Exception("unable to extract CVM IP from node {}", node)
         return cvm_ip
 
+    def _verify_discovered_nodes_imaging_spec(self, node_spec):
+        # required params check for discovered nodes
+        required_params = ["hypervisor_hostname","hypervisor_ip", "cvm_ip", "ipmi_ip"]
+
+        # discovery os based nodes have "pheonix" value for hypervisor and its not valid hypervisor for imaging
+        hypervisor = node_spec.get("hypervisor")
+        if (hypervisor is None) or (hypervisor == "pheonix") :
+            #return error
+            return "Please provide valid hypervisor in discovery_override"
+
+        for param in required_params:
+            if node_spec.get(param):
+                pass
+            else:
+                # return error
+                return "{} value didn't retrieved while discovery, please provide it in discovery_override".format(param)
```

Comment:
Change the error message
"Missing params: [x, y, z] or value not set. 
We have strip_attribs_with_none_values() that actually strips all params which is set as None. 
Why do you need this check here? is nodes_spec not part of ansible_params?

Reply:
This params are overrided by node network details. In scenarios like dos theres no hypervisor details or any other details missing after discovery needs to be validated after discovery only.

Reply:
Updated the comment